packages/contracts/contracts/optimistic-ethereum/ovm/ExecutionManager.sol

```diff
@@ -665,11 +651,8 @@ contract ExecutionManager {
      * @param _ovmInitcode The initcode for our new contract
      * @return True if this succeeded, false otherwise.
      */
-    function createNewContract(address _newOvmContractAddress, bytes memory _ovmInitcode) internal returns (bool){
-        if (!safetyChecker.isBytecodeSafe(_ovmInitcode)) {
-            // Contract init code is not safe.
-            return false;
-        }
+    function createNewContract(address _newOvmContractAddress, bytes memory _ovmInitcode) internal {
+        require(safetyChecker.isBytecodeSafe(_ovmInitcode), "Contract init (creation) code is not safe");
```

Comment:
Will we still need to return the address it _would have been_ deployed to in this case? If so, this will not do that, right?

Reply:
yep, it will revert if the code is not safe. Even though this differs from evm CREATE/CREATE2, which always returns the address it would have been deployed to, I think it's worth it for ease of developer debugging. If my bytecode is failing to deploy, it would be ideal as a dev to know that it's because I forgot to transpile it first, not because of some OVM bug.